core/trino-main/src/main/java/io/trino/operator/aggregation/listagg/ListaggAggregationFunction.java

```diff
@@ -0,0 +1,259 @@
+                        stateInterface,
+                        stateSerializer,
+                        stateFactory)),
+                outputType);
+
+        GenericAccumulatorFactoryBinder factory = AccumulatorCompiler.generateAccumulatorFactoryBinder(metadata, classLoader);
+        return new InternalAggregationFunction(NAME, inputTypes, ImmutableList.of(intermediateType), outputType, factory);
+    }
+
+    public static void input(Type type, ListaggAggregationState state, Block value, Slice separator, boolean overflowError, Slice overflowTruncationFiller, boolean overflowTruncationCountIndication, int position)
+    {
+        if (state.isEmpty()) {
+            if (overflowTruncationFiller.length() > MAX_TRUNCATION_FILLER_LENGTH) {
+                throw new TrinoException(INVALID_FUNCTION_ARGUMENT, format("truncation filler length %d exceeds maximum length %d", overflowTruncationFiller.length(), MAX_TRUNCATION_FILLER_LENGTH));
+            }
+            // Set the parameters of the LISTAGG command within the state so that
+            // they can be used within the `output` function
+            state.setSeparator(separator);
+            state.setOverflowError(overflowError);
+            state.setOverflowTruncationFiller(overflowTruncationFiller);
+            state.setOverflowTruncationCountIndication(overflowTruncationCountIndication);
+        }
+        if (!value.isNull(position)) {
+            state.add(value, position);
+        }
+    }
+
+    public static void combine(Type type, ListaggAggregationState state, ListaggAggregationState otherState)
+    {
+        Slice previousSeparator = state.getSeparator();
+        if (previousSeparator == null) {
+            state.setSeparator(otherState.getSeparator());
+            state.setOverflowError(otherState.isOverflowError());
+            state.setOverflowTruncationFiller(otherState.getOverflowTruncationFiller());
+            state.setOverflowTruncationCountIndication(otherState.isOverflowTruncationCountIndication());
+        }
+
+        state.merge(otherState);
+    }
+
+    public static void output(Type type, ListaggAggregationState state, BlockBuilder out)
+    {
+        if (state.isEmpty()) {
+            out.appendNull();
+        }
+        else {
+            outputState(state, out, MAX_OUTPUT_LENGTH);
+        }
+    }
+
+    @VisibleForTesting
+    protected static void outputState(ListaggAggregationState state, BlockBuilder out, int maxOutputLength)
+    {
+        Slice separator = state.getSeparator();
+        int separatorLength = separator.length();
+        OutputContext outputContext = new OutputContext();
+        state.forEach((block, position) -> {
+            if (outputContext.aggregationBytesLength > maxOutputLength) {
+                if (state.isOverflowTruncationCountIndication()) {
+                    outputContext.aggregationBytesLength += separatorLength + block.getSliceLength(position);
+                }
+            }
```

Comment:
What's the purpose of this? Once we know it overflows, we should just output the result. No point in tracking the length any further.

Reply:
I think it would be beneficial for the user to receive an information in case of an overflow error on how big would be the actual aggregation length.

Reply:
I don't think it's that important to do so. Just telling the the maximum size was exceeded is sufficient. Tracking the length also requires unnecessary processing for the case where the user doesn't want it to fail on overflow.